Plugin Throws Error when Creating VM with GPU

Plugin Version:          3.0.2-rc05
Proxmox Version:      9.0.10
OpenTofu Version:    1.10.6
---

Hi Guys, I get the following error when I try to provision a vm that has a gpu mapping, The VM seems to create and start fine with the GPU mapped so not sure what's causing the error. If I remove the `pci0{..}` statement it seems to work fine.

I did notice in the webui it also shows this error, which might be related somehow? 
```
VM 100 qmp command 'set_password' failed - unable to connect to VM 100 qmp socket - timeout after 51 retries
TASK ERROR: Failed to run vncproxy.
```

Plugin Error

```
Stack trace from the terraform-provider-proxmox_v3.0.2-rc05 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa0d94d]

goroutine 23 [running]:
github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/pci.terraformSubroutinePCIs({0x0?, 0xc0010bceb0?, 0x0?})
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/pci/terraform.go:175 +0x20d
github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/pci.terraformPCIs(0xc001258db0)
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/pci/terraform.go:154 +0xe5
github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/pci.Terraform(0xc001258db0, 0xc000e5a090)
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/qemu/pci/terraform.go:17 +0xbf
github.com/Telmate/terraform-provider-proxmox/v2/proxmox.resourceVmQemuRead({0x108b6d8, 0xc0002200e0}, 0xc000e5a090, {0xd6dea0?, 0xc0007f2fa0})
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/resource_vm_qemu.go:1071 +0x15ff
github.com/Telmate/terraform-provider-proxmox/v2/proxmox.resourceVmQemuCreate({0x108b6d8, 0xc0002200e0}, 0xc000e5a090, {0xd6dea0, 0xc0007f2fa0})
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/resource_vm_qemu.go:784 +0x3245
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc00016d900, {0x108b630, 0xc000ba33e0}, 0xc000e5a090, {0xd6dea0, 0xc0007f2fa0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.37.0/helper/schema/resource.go:849 +0x119
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00016d900, {0x108b630, 0xc000ba33e0}, 0xc000814ea0, 0xc000d1ddd0, {0xd6dea0, 0xc0007f2fa0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.37.0/helper/schema/resource.go:980 +0xb47
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00038c030, {0x108b630?, 0xc000ba3320?}, 0xc0007f2410)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.37.0/helper/schema/grpc_provider.go:1484 +0x1085
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0002960a0, {0x108b630?, 0xc000ba21e0?}, 0xc0001a0c80)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov5/tf5server/server.go:941 +0x3b9
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xefdb20, 0xc0002960a0}, {0x108b630, 0xc000ba21e0}, 0xc0001a0b80, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:687 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000276400, {0x108b630, 0xc000ba2150}, 0xc0007fe180, 0xc00098b140, 0x1729fe8, 0x0)
        google.golang.org/grpc@v1.72.1/server.go:1405 +0x1036
google.golang.org/grpc.(*Server).handleStream(0xc000276400, {0x108c010, 0xc0000fa000}, 0xc0007fe180)
        google.golang.org/grpc@v1.72.1/server.go:1815 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.72.1/server.go:1035 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 52
        google.golang.org/grpc@v1.72.1/server.go:1046 +0x11d

Error: The terraform-provider-proxmox_v3.0.2-rc05 plugin crashed!
```

--- a/proxmox/Internal/resource/guest/qemu/pci/terraform.go
+++ b/proxmox/Internal/resource/guest/qemu/pci/terraform.go
@@ -172,7 +172,7 @@ func terraformSubroutinePCIs(config pveAPI.QemuPci) []interface{} {
 			params[schemaPrimaryGPU] = *config.Mapping.PrimaryGPU
 		}
 		if config.Mapping.MDev != nil {
-			params[schemaMDev] = config.Mapping.VendorID.String()
+			params[schemaMDev] = config.Mapping.MDev.String()
 		}
 		if config.Mapping.ROMbar != nil {
 			params[schemaROMbar] = *config.Mapping.ROMbar

